pkg/pool-manager/pool_test.go

```diff
@@ -589,4 +590,223 @@ var _ = Describe("Pool", func() {
 			})
 		})
 	})
+
+	type compareNamespaceSelectorToOptModeParams struct {
+		mutatingWebhookConfigurationName string
+		namespaceName                    string
+		optMode                          OptMode
+		ErrorTextExpected                string
```

Comment:
I would not insist on this, but IMO it is wrong to assert on error text.
You should assert on the actual error object. The implication is that the errors are either pre-defined using `New` or you use a specific type check that is exposed.
Then the assertion is against a specific instance (if you exposed the error using New) or against a specific type (in case you expose the error type).

Reply:
I think this method does not go hand in hand with the `errors.wrap()` method, where all the errors are concatenated with `:` between them. That's why I prefer to match the error text.

I changed the test to use gomega's matchError that supports error text matching.

Reply:
It is the opposite, the `Wrap` is exposing the instance and type object of the inner error, not the text. Text exposure does not need to use wrapping, a simple `%v` does the job.
You can read about it [here](https://blog.golang.org/go1.13-errors).